test/instrumentation/modules/memcached.js

```diff
@@ -0,0 +1,52 @@
+'use strict'
+
+var agent = require('../../..').start({
+  serviceName: 'test',
+  secretToken: 'test',
+  captureExceptions: false,
+  metricsInterval: 0
+})
+
+var test = require('tape')
+var mockClient = require('../../_mock_http_client')
+test(function (t) {
+  resetAgent(function (data) {
```

Comment:
Within this callback, you need to check that the events sent by the agent are as expected.

All the events will be contained in the `data` argument, which will have the following format:

```js
{
  length: 8,
  spans: [...],
  transactions: [...],
  errors: [...]
}
```

You can see an example of how we validate this in the handlebar tests:

https://github.com/elastic/apm-agent-nodejs/blob/a907f573864da7e391491622012642ad811cac5b/test/instrumentation/modules/handlebars.js#L19-L39